Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -280,6 +281,19 @@ class TIFFLoadingContext {
             TRY(loop_over_pixels(move(decode_lzw_strip)));
             break;
         }
+        case Compression::AdobeDeflate: {
+            // This is an extension from the Technical Notes from 2002:
+            // https://web.archive.org/web/20160305055905/http://partners.adobe.com/public/developer/en/tiff/TIFFphotoshop.pdf
+            ByteBuffer decoded_bytes {};
+            auto decode_zlib = [&](u32 num_bytes) -> ErrorOr<ReadonlyBytes> {
+                auto const encoded_bytes = TRY(m_stream->read_in_place<u8 const>(num_bytes));
+                decoded_bytes = TRY(Compress::ZlibDecompressor::decompress_all(encoded_bytes));
```

Comment:
However, since I generally dislike the implications of a `decompress_all` method (even though we eliminate one copy of the data via `read_in_place`), I'd suggest to just replace this with the following (conceptually, as this almost certainly won't work as-is since I'm phone-posting) and entirely skip adding `ZlibDecompressor::decompress_all`:

```suggestion
                auto stream = TRY(ConstrainedStream::create(*m_stream, num_bytes));
                auto decompressed_stream = TRY(Compress::ZlibDecompressor::create(move(stream)));
                decoded_bytes = TRY(decompressed_stream->read_until_eof(4096));
```

Reply:
I didn't know we have `ConstrainedStream`, very nice!